### DISCUSSION TITLE
Sending A GTM event to the dataLayer on a successfull donation

### DISCUSSION THREAD

**Discussion by hanbga** [impress-org/givewp#7318] [Developer Q&A]
Hello, I'm currently utilizing the give_get_success_page_uri filter within the legacy template. This allows me to obtain donation data and send it to the dataLayer for additional processing. However, I've encountered issues where this filter doesn't function with Multistep forms, Classic Forms, and the new form builder. What would be the most effective approach to address this across these various scenarios?

**Comment by jonwaldstein** [comment]
Hey @hanbga 

Thanks for reaching out!

Can you share how exactly you are using the filter?  That context should help me guide a direction better.

Thanks!

**Reply by hanbga** [reply]
add_filter('give_get_success_page_uri', 'give_add_donation_parameters_to_url', 10, 2);

function give_add_donation_parameters_to_url($uri)
{
    $donation_amount = isset($_POST['give-amount']) ? $_POST['give-amount'] : 0;
    $form_name = isset($_POST['give-form-title']) ? $_POST['give-form-title'] : 'default-form';

    // Get user email from the GiveWP form
    $user_email = isset($_POST['give_email']) ? $_POST['give_email'] : '';

    // Add user email to the URL if available
    if ($user_email) {
        $uri = add_query_arg('email', $user_email, $uri);
    }

    // Add other parameters as needed
    $uri = add_query_arg('amount', $donation_amount, $uri);
    $uri = add_query_arg('form', $form_name, $uri);

    return $uri;
}
sample of me adding this to the donation confirmation url. then on the confirmation page I can pick the data and send the GTM event using js. But this filter doesnt run at all on other templates because they do not have the thank you page functionality

**Reply by hanbga** [reply]
@jonwaldstein

**Reply by jonwaldstein** [reply]
Hey @hanbga 

So yes, you are correct - what you are currently trying to accomplish is not possible because the confirmation page is generated between different form templates and form types.  Also, the concept of a successful donation varies between gateways being used. We have talked about this in the past and had some ideas but do not have anything concrete yet.

However, a little bit more context from you might help: Are you hoping to track when a donation form is considered completed and lands on the confirmation page?

**Reply by hanbga** [reply]
I'm trying to send successful donation data to the dataLayer (GTM) for various analytics pixels for tracking. What is the best way to do this using the new form templates.

**Reply by hanbga** [reply]
@jonwaldstein

**Reply by jonwaldstein** [reply]
@hanbga there are ways on the server to determine when a donation was successful by listening for its status to change to complete, however I do not have a recommended way to do this on the front-end currently.

**Reply by iwilliamsLF** [reply]
Hey @hanbga wondering if you ever found a solution for this issue? Or @jonwaldstein have there been any developments that would assist with this feature?

**Reply by jonwaldstein** [reply]
In visual builder forms there is the ability to redirect the confirmation page to the one defined in give settings.  So perhaps you could add the event to that page:

<img width="1512" height="839" alt="Screenshot 2025-12-05 at 3 17 23 PM" src="https://github.com/user-attachments/assets/ee8bb88d-519a-465b-be16-174ce2c8d3eb" />
